src/switch_rtp.c

```diff
@@ -7607,7 +7607,6 @@ static switch_status_t process_rtcp_xr_report(switch_rtcp_report_data_t *report_
 		return status;
 	}
 
-	report_data->xr_count = count;
 	switch_malloc(report_data->rtcp_data.xr_blocks, sizeof(switch_rtcp_xr_report) * count);
```

Comment:
What if all xr reports is invalid or unsupported? I think you free up blocks based on the number of xr_count. So if xr_count is zero then will not be able to clean up the xr_blocks which will cause a mem leak

Reply:
good catch! thanks